nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkManagerImpl.java

```diff
@@ -450,11 +453,15 @@ protected boolean shutdownExecutors(Collection<WorkThreadPoolExecutor> list, lon
             if (!executor.awaitTermination(timeout, TimeUnit.MILLISECONDS)) {
                 return false;
             }
-            timeout -= unit.convert(System.currentTimeMillis() - t0, TimeUnit.MILLISECONDS);
+            timeout -= System.currentTimeMillis() - t0;
```

Comment:
This is old code (modulo the unit bug you fixed) but we should also check we don't go into negative values to be robust. If timeout becomes <= 0 we should return false as well.

Reply:
The code called, awaitTermintation(timeout, unit), handles it already, I believe it's OK to let this part as is since it would need to be tested twice and would get uglier.